core/headerchain.go

```diff
@@ -80,22 +82,46 @@ type HeaderChain struct {
 
 // NewHeaderChain creates a new HeaderChain structure. ProcInterrupt points
 // to the parent's interrupt semaphore.
-func NewHeaderChain(db ethdb.Database, engine consensus.Engine, pEtxsRollupFetcher getPendingEtxsRollup, pEtxsFetcher getPendingEtxs, chainConfig *params.ChainConfig, cacheConfig *CacheConfig, txLookupLimit *uint64, indexerConfig *IndexerConfig, vmConfig vm.Config, slicesRunning []common.Location, logger *log.Logger) (*HeaderChain, error) {
+func NewHeaderChain(db ethdb.Database, engine consensus.Engine, pEtxsRollupFetcher getPendingEtxsRollup, pEtxsFetcher getPendingEtxs, chainConfig *params.ChainConfig, cacheConfig *CacheConfig, txLookupLimit *uint64, indexerConfig *IndexerConfig, vmConfig vm.Config, slicesRunning []common.Location, currentExpansionNumber uint8, logger *log.Logger) (*HeaderChain, error) {
 	headerCache, _ := lru.New(headerCacheLimit)
 	numberCache, _ := lru.New(numberCacheLimit)
 	nodeCtx := chainConfig.Location.Context()
 
 	hc := &HeaderChain{
-		config:          chainConfig,
-		headerDb:        db,
-		headerCache:     headerCache,
-		numberCache:     numberCache,
-		engine:          engine,
-		slicesRunning:   slicesRunning,
-		fetchPEtxRollup: pEtxsRollupFetcher,
-		fetchPEtx:       pEtxsFetcher,
-		logger:          logger,
-		indexerConfig:   indexerConfig,
+		config:                 chainConfig,
+		headerDb:               db,
+		headerCache:            headerCache,
+		numberCache:            numberCache,
+		engine:                 engine,
+		slicesRunning:          slicesRunning,
+		fetchPEtxRollup:        pEtxsRollupFetcher,
+		fetchPEtx:              pEtxsFetcher,
+		logger:                 logger,
+		indexerConfig:          indexerConfig,
+		currentExpansionNumber: currentExpansionNumber,
+	}
+
+	genesisHash := hc.GetGenesisHashes()[0]
+	hc.genesisHeader = hc.GetHeaderByHash(genesisHash)
+	if bytes.Equal(chainConfig.Location, common.Location{0, 0}) {
+		// TODO: Need a check to match the hash, this is not the right place for this but have to
```

Comment:
Is this done?

Reply:
yeah, this is for checking the genesis hash mismatch